test/plays/service/PlaysService.spec.ts

```diff
@@ -139,11 +176,13 @@ describe('Plays Service', () => {
       expect(actualPlay.name).to.equal(expectedPlay.name);
       expect(actualPlay.createdAt.toISOString()).to.equal(expectedPlay.createdAt.toISOString());
       expect(actualPlay.updatedAt.toISOString()).to.equal(expectedPlay.updatedAt.toISOString());
-      expect(actualPlay.steps[0].environmentVariableKeys).to.equal(expectedPlay.steps[0].environmentVariableKeys);
-      expect(actualPlay.steps[0].id).to.equal(expectedPlay.steps[0].id);
-      expect(actualPlay.steps[0].name).to.equal(expectedPlay.steps[0].name);
-      expect(actualPlay.steps[0].createdAt.toISOString()).to.equal(expectedPlay.steps[0].createdAt.toISOString());
-      expect(actualPlay.steps[0].updatedAt.toISOString()).to.equal(expectedPlay.steps[0].updatedAt.toISOString());
+      if (actualPlay.steps && expectedPlay.steps) {
```

Comment:
@ManojKumarChannelApe Right now I believe the following scenarios would result in a passing test, even though it should fail:

- `actualPlay.steps` has a value, but `expectedPlay.steps` is undefined
- `expectedPlay.steps` has a value, but `actualPlay.steps` is undefined

I think you just need to perform some check(s) here to ensure that if `actualPlay.steps` has a value, so does `expectedPlay.steps`; and vice versa.

The actual tests which call `expectPlay()` look fine, I think it would be safer to cover this in case the tests are changed in the future or new tests are added which use this method.

Reply:
@mb-ca I have already checked in expectPlay method for actualPlay.steps and expectedPlay.steps if they have value or not

Reply:
@mb-ca then we can add one more test case and check like this   expect(expectedPlay.steps).to.not.equal(actualPlay2.steps) . let me know if you disagree .?

Reply:
@ManojKumarChannelApe No need to add another test case, just update the expectPlay function with the following:
```typescript
function expectPlay(actualPlay: Play) {
	expect(actualPlay.id).to.equal(expectedPlay.id);
	expect(actualPlay.name).to.equal(expectedPlay.name);
	expect(actualPlay.createdAt.toISOString()).to.equal(expectedPlay.createdAt.toISOString());
	expect(actualPlay.updatedAt.toISOString()).to.equal(expectedPlay.updatedAt.toISOString());

	if (expectedPlay.steps) {
		if (!actualPlay.steps) {
			fail('Expected actual play result to have steps');
		} else {
			expect(actualPlay.steps.length).to.equal(expectedPlay.steps.length);

			expect(actualPlay.steps[0].environmentVariableKeys).to.equal(expectedPlay.steps[0].environmentVariableKeys);
			expect(actualPlay.steps[0].id).to.equal(expectedPlay.steps[0].id);
			expect(actualPlay.steps[0].name).to.equal(expectedPlay.steps[0].name);
			expect(actualPlay.steps[0].createdAt.toISOString()).to.equal(expectedPlay.steps[0].createdAt.toISOString());
			expect(actualPlay.steps[0].updatedAt.toISOString()).to.equal(expectedPlay.steps[0].updatedAt.toISOString());
		}
	} else {
		expect(actualPlay.steps).to.be.undefined;
	}
}
```